src/main/java/seedu/address/model/card/Card.java

```diff
@@ -22,12 +23,12 @@ public class Card {
     // Data fields
     private final Answer answer;
     private final Set<Tag> tags = new HashSet<>();
-    private final Deck deck;
+    private Optional<Deck> deck;
 
     /**
      * Every field must be present and not null.
      */
-    public Card(Question question, Answer answer, Set<Tag> tags, Deck deck) {
+    public Card(Question question, Answer answer, Set<Tag> tags, Optional<Deck> deck) {
```

Comment:
I dont think the Card constructor should have a deck as optional. We should require non-null so that a Card can only be created if it is in a Deck